hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/utils/SparkMetadataWriterUtils.java

```diff
@@ -190,9 +190,16 @@ private static List<HoodieRecord> getUnmergedLogFileRecords(List<String> logFile
         .withLatestInstantTime(metaClient.getActiveTimeline().getCommitsTimeline().lastInstant().get().requestedTime())
         .withReaderSchema(readerSchema)
         .withTableMetaClient(metaClient)
-        .withLogRecordScannerCallback(records::add)
         .build();
-    scanner.scan(false);
+    Iterator<HoodieRecord<?>> recordIterator = scanner.iterator();
```

Comment:
Is it possible we return an iterator too for method `getUnmergedLogFileRecords` to the list add/get can be simplified.